### PR TITLE
migrate ZooModel to support 4dmm for TimeDistributed,TransformerLayer,Bert

### PR DESCRIPTION
**Pull request by Le-Zheng**

### CODE REVIEW

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/models/common/ZooModel.scala [hunk 1]
```diff
@@ -0,0 +1,154 @@
+/*
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/models/common/ZooModel.scala:1]: why do we need this class? where is it used?

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/layers/TimeDistributed.scala [hunk 2]
```diff
@@ -22,89 +22,88 @@ import com.intel.analytics.bigdl.dllib.keras.{KerasLayer}
 import com.intel.analytics.bigdl.dllib.tensor.Tensor
 import com.intel.analytics.bigdl.dllib.tensor.TensorNumericMath.TensorNumeric
 import com.intel.analytics.bigdl.dllib.utils.Shape
-// import com.intel.analytics.zoo.models.common.ZooModel
+import com.intel.analytics.bigdl.dllib.models.common.ZooModel
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/layers/TimeDistributed.scala:25]: why do we need this?

Comment (Le-Zheng) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/layers/TimeDistributed.scala:25]: > why do we need this?

I found the chatbot example uses TimeDistributed. https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/examples/chatbot/Train.scala#L128

Comment (jason-dai) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/layers/TimeDistributed.scala:25]: But `Dense` is not a `ZooModel`, yes?

Comment (Le-Zheng) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/layers/TimeDistributed.scala]: > But `Dense` is not a `ZooModel`, yes?

Yes. I will not migrate ZooModel and remove the ZooModel part in TimeDistributed.scala.

**Comment by Le-Zheng** [comment]
Close this pr, and move changes in next pr https://github.com/intel-analytics/analytics-zoo/pull/4736

**Pull request closed without merging**